src/radical/pilot/agent/launch_method/base.py

```diff
@@ -53,57 +53,34 @@
 #
 class LaunchMethod(object):
 
-    # List of environment variables that designated Launch Methods should export
-    # FIXME: we should find out what env vars are changed or added by
-    #        td.pre_exec, and then should also export those.  That would make
-    #        our launch script ore complicated though...
-    EXPORT_ENV_VARIABLES = [
-        'LD_LIBRARY_PATH',
-        'PATH',
-        'PYTHONPATH',
-        'OMP_NUM_THREADS',
-        'CUDA_VISIBLE_DEVICES',
-        'RP_AGENT_ID',
-        'RP_GTOD',
-        'RP_PILOT_ID',
-        'RP_PILOT_STAGING',
-        'RP_PROF',
-        'RP_SESSION_ID',
-        'RP_SPAWNER_ID',
-        'RP_TMP',
-        'RP_TASK_ID',
-        'RP_TASK_NAME',
-        'RP_PILOT_SANDBOX',
-        'RADICAL_BASE'
-    ]
-
     MPI_FLAVOR_OMPI    = 'OMPI'
     MPI_FLAVOR_HYDRA   = 'HYDRA'
     MPI_FLAVOR_UNKNOWN = 'unknown'
 
+
     # --------------------------------------------------------------------------
     #
-    def __init__(self, name, cfg, session):
+    def __init__(self, name, cfg, log, prof):
 
-        self.name     = name
-        self._cfg     = cfg
-        self._session = session
-        self._log     = self._session._log    # pylint: disable=protected-access
-        self._log.debug('create LaunchMethod: %s', type(self))
+        log.debug('===== lm base init start')
+        self.name  = name
+        self._cfg  = cfg
+        self._log  = log
+        self._prof = prof
+        self._pwd  = os.getcwd()
 
-        # A per-launch_method list of env vars to remove from the Task env
-        self.env_removables = []
-
-        self._configure()
+        self._info = self._cfg.get('lm_info', {}).get(self.name)
```

Comment:
- inside RM, this attribute will be empty here, but will be set during `initialize` method (from provided config file), and structures inside `self._info` for these cases will be different
- inside Executor, this attribute will be set here, but will not be used anyhow, since configuration was done within `initialize` method

these two points are in favor of what I wrote earlier: to split LM class in a way that one [sub-class] will be used in RM and another in Executor; to have configuration procedure within Executor.